embedding/embedlite/utils/WebBrowserChrome.cpp

```diff
@@ -256,12 +253,16 @@ WebBrowserChrome::OnStateChange(nsIWebProgress* progress, nsIRequest* request,
   }
 
   if (progressStateFlags & nsIWebProgressListener::STATE_START && progressStateFlags & nsIWebProgressListener::STATE_IS_DOCUMENT) {
-    Unused << AddUserAgentObserver(request);
     mListener->OnLoadStarted(mLastLocation.get());
   }
   if (progressStateFlags & nsIWebProgressListener::STATE_STOP && progressStateFlags & nsIWebProgressListener::STATE_IS_DOCUMENT) {
-    Unused << RemoveUserAgentObserver(request);
     mListener->OnLoadFinished();
+    nsAutoString httpUserAgent;
+    nsresult rv = GetHttpUserAgent(request, httpUserAgent);
+    if (NS_SUCCEEDED(rv) && !httpUserAgent.IsEmpty()) {
```

Comment:
What if the user-agent override is really empty? Pondering do we need that is not empty guard.

Reply:
During my testing msn.com sometimes loaded about:blank too which had empty ua, added this guard to prevent overwrite.

Reply:
Yeap, there's about:blank when new tab request start. So, that's expected. Thanks for clarifying => let's leave it there.